pkg/cloud/services/networking/securitygroups.go

```diff
@@ -200,7 +202,29 @@ func (s *Service) generateDesiredSecGroups(openStackCluster *infrav1.OpenStackCl
 	workerRules := append([]resolvedSecurityGroupRuleSpec{}, defaultRules...)
 
 	controlPlaneRules = append(controlPlaneRules, getSGControlPlaneHTTPS()...)
-	workerRules = append(workerRules, getSGWorkerNodePort()...)
+
+	// Fetch subnet to use for worker node port rules
+	// In the future IPv6 support need to be added here
+	if openStackCluster.Status.Network != nil && len(openStackCluster.Status.Network.Subnets) > 0 {
+		for _, subnet := range openStackCluster.Status.Network.Subnets {
+			// Check uf subnet.CIDR is ipv4 subnet
+			_, ipnet, err := net.ParseCIDR(subnet.CIDR)
+			if err != nil {
+				return nil, fmt.Errorf("invalid subnet found during security groups reconcile: %v", err)
+			}
+			if ipnet.IP.To4() != nil {
+				SubnetCIDR = subnet.CIDR
+				break
+			}
+		}
+	}
+	if SubnetCIDR != "" {
+		// If SubnetCIDR is found we allow tcp and udp traffic from said SubnetCIDR, this in order to allow octavia Loadbalancers created by CCM to function properly
+		workerRules = append(workerRules, getSGWorkerNodePortCidr(SubnetCIDR)...)
```

Comment:
You could move this inside the block above and also remove the break so you simply add any listed IPv4 subnet. Any other restrictions we have on subnets aren't relevant here, so might as well make it handle whatever's there.